runtime/interpreter/decode.go

```diff
@@ -1099,61 +1276,74 @@ func (d *DecoderV4) decodeLink(v interface{}) (LinkValue, error) {
 	}, nil
 }
 
-func (d *DecoderV4) decodeStaticType(v interface{}) (StaticType, error) {
-	tag, ok := v.(cbor.Tag)
-	if !ok {
-		return nil, fmt.Errorf("invalid static type encoding: %T", v)
-	}
+func (d *DecoderV4) decodeStaticType() (StaticType, error) {
+	number, err := d.decoder.DecodeTagNumber()
+	if err != nil {
+		if e, ok := err.(*cbor.WrongTypeError); ok {
+
+			// CBOR nil is interpreted as nil static type.
+			if e.ActualType == cbor.NilType {
+				if err := d.decoder.Skip(); err != nil {
+					return nil, err
+				}
+				return nil, nil
+			}
```

Comment:
I think this answers my question above, if the capability borrow type is still optional.

The static type in other values is often optional, but not always, e.g. it is required for links.

I think it would be good to keep `decodeStaticType` to be non-optional and check for `cbor.NilType` where it could be optional

Reply:
I updated PR to premove optional static type decoding from decodeStaticType(). That is now done in 3 places which need optional static type decoding.

Reply:
4d8b876 looks good!